plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/validation/TypeValidator.java

```diff
@@ -165,6 +167,24 @@ public void setBase(BaseValidator base) {
 			}
 		});
 		
+		
+		
+		clazz.getAttributes().stream()
+			.filter(att -> att.getFeatureRef().getEType() == ImplementationPackage.eINSTANCE.getUnresolvedEClassifier())
+			.forEach(att -> {
+				String declaredPackages = base.qryEnv.getEPackageProvider().getRegisteredEPackages().stream()
+						.map(p -> p.getName())
+						.collect(joining(", ","[","]")); 		
+				msgs.add(new ValidationMessage(
+						ValidationMessageLevel.ERROR,
+						// TODO implement a contextual UnresolvedType to get better name. cf. https://github.com/gemoc/ale-lang/issues/78
+						String.format(UNRESOLVED_TYPE_EXT, ""/*getQualifiedName(att.getFeatureRef().getEType())*/,declaredPackages),
+						base.getStartOffset(att),
+						base.getEndOffset(att)
+						));
+				
```

Comment:
Looks like this piece of code is duplicated at line 681. Maybe extract it as a method?

Combined with the previous comment it could also enhance code's readability:

```java
clazz.getAttributes().stream()
     .filter(attr -> hasAnUnresolvedType(attr))
     .forEach(untypedAttr -> createUnresolvedTypeErrorMessage(untypedAttr));
```